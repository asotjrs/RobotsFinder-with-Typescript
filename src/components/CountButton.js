import React, {Component} from "react";

class CountButton extends Component{

    constructor(props){
        super(props);
        this.state={
            count:0
        }
    }

    updateCount=()=>{

        this.setState(state=>{
            return {count:state.count+1}
        })
    };


    render() {
        return<button id="counter" color={this.props.color} onClick={this.updateCount}>Count {this.state.count}</button>
    }


}
export default  CountButton;