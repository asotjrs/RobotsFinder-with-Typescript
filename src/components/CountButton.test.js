import React from "react";
import {shallow} from 'enzyme';
import toJson from "enzyme-to-json";
import CountButton from "./CountButton";
it('should render a Buttoncount', function () {
    expect(toJson(shallow(<CountButton/>))).toMatchSnapshot();
});
it('should simulate an onclick', function () {
    const mockColor="red";
    const wrapper=shallow(<CountButton color={mockColor}/>);
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count:1});
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count:2});
    expect(wrapper.props().color).toEqual("red");
});