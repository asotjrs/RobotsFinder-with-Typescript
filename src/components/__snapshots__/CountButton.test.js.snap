// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Buttoncount 1`] = `
<button
  id="counter"
  onClick={[Function]}
>
  Count 
  0
</button>
`;
